### PR TITLE
Add sanitization for custom email settings string

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/5776 

image

https://user-images.githubusercontent.com/12435965/180504782-092a0df0-a00a-41fc-a953-658bbc4431da.mp4

### CODE REVIEW

File: server/lib/collectivelib.ts [hunk 1]
```diff
@@ -148,11 +149,10 @@ export function validateSettings(settings: any): string | boolean {
...
   /*
-   * Validate customEmailMessage length. We are using 511 characters to account
-   * for the "div" tags that enclose the content <div>..<div/>. These divs are auto
-   * added by the RichTextEditor.js
+   * Validate customEmailMessage length.
    */
-  if (settings.customEmailMessage && settings.customEmailMessage.length > 511) {
+  if (settings.customEmailMessage && stripHTML(settings.customEmailMessage).length > 500) {
+    console.log(stripHTML(settings.customEmailMessage));
     return 'Message should be less than 500 characters';
```
Comment (Betree) [inline server/lib/collectivelib.ts:156]: Would be great to be more explicit on this one.

```suggestion
    return 'Custom "Thank you" email message should be less than 500 characters';
```

File: server/lib/search.js [hunk 2]
```diff
@@ -178,7 +178,7 @@ export const searchCollectivesInDB = async (
   }
 
   if (tags?.length) {
-    searchedTags = `{${tags.map(tag => `"${tag}"`).join(',')}}`;
+    searchedTags = `{${tags.map(tag => `"${tag.toLowerCase()}"`).join(',')}}`;
```
Comment (Betree) [inline server/lib/search.js:181]: I don't get why we're changing this here?

**Review by Betree — COMMENTED** [review]
Looks like the right approach but there are a few things to resolve still

File: server/lib/search.js (hunk 2 above)
Comment (SudharakaP) [reply server/lib/search.js:181]: Sorry about this one, I was trying this out for a different issue and somehow accidentally checked in.

File: server/graphql/v2/mutation/AccountMutations.ts [hunk 3]
```diff
@@ -99,7 +115,8 @@ const accountMutations = {
         }
 
         const settings = account.settings ? cloneDeep(account.settings) : {};
-        set(settings, args.key, args.value);
+        const value = sanitizeSettingsValue(args.key, args.value);
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply server/graphql/v2/mutation/AccountMutations.ts]: Ah, makes sense. I've moved it to the Collective mode. 👍🏽

**Review by Betree — COMMENTED** [review]
Tests are failing because `settings` can be null.

I've added an additional test to make sure the feature works.

**Comment by SudharakaP** [comment]
> Tests are failing because `settings` can be null.
> 
> I've added an additional test to make sure the feature works.

Thanks for the review and the additional check. I've fixed the problem. 👍🏽

**Review by Betree — APPROVED** [review]

**Pull request merged**